src/cadenza.js

```diff
@@ -54,6 +54,11 @@ globalThis.cadenza = Object.assign(
 /** @typedef {EmbeddingTargetId | WorksheetKey} WorksheetSource - A worksheet source */
 /** @typedef {EmbeddingTargetId | WorkbookViewKey} WorkbookViewSource - A workbook view source */
 
+/**
+ * @typedef PageSource - A Cadenza landing page
+ * @property {string} page - The name of the landing page
```

Comment:
string -> 'welcome'

Reply:
I do not understand what you mean by this.

Reply:
Oh, got it now. But I really do not think it should be the compiler error. It should be a runtime error, as it happens for all other arg validations we have in the library.

Also, why directly limit this to 'welcome'? In theory the api should support other pages.

Reply:
> It should be a runtime error
Why would you let people call this method if we know at build time that the argument would be wrong? That's the point of TS, right?

> as it happens for all other arg validations we have in the library.
The validation for the other cases are also on top of maximum specific types. The validations add additional safety for when we cannot express certain things as types or for users that do not use TS.

> Also, why directly limit this to 'welcome'? In theory the api should support other pages.
If we would add another well-known page, we'd also extend the validation.

Reply:
That is the point of TS indeed but this is JS, so we're still doing things like validateGeometryType. And we even have unit tests for these validations, so they are not expected to be compile errors, 'cause that would not even allow us to write a unit test.

If I do the suggested change I get:
{code}
    src/cadenza.test.ts:198:16 - error TS2345: Argument of type '{ page: string; }' is not assignable to parameter of type 'string | PageSource | WorkbookKey | WorksheetKey | WorkbookViewKey'.
      Type '{ page: string; }' is not assignable to type 'PageSource'.
        Types of property 'page' are incompatible.
          Type 'string' is not assignable to type '"welcome"'.

    198       cad.show(WELCOME_PAGE);
                       ~~~~~~~~~~~~
{code}

Why should we insist doing for page source something differently that has been done for media type and geometry type?

Reply:
For MediaType and GeometryType we also have @typedefs. Based on these typedefs we generate a .d.ts file for people who use typescript.

Reply:
I fixed the previous compilation error by importing the PageSource type, so we should be good to go.